pyccl/core.py

```diff
@@ -194,6 +194,23 @@ class Cosmology(object):
                                      "HMCode_logT_AGN": 7.8}}
 
     """
+    # Go through all functions in the main package and the subpackages
+    # and make every function that takes `cosmo` as its first argument
+    # an attribute of this class.
+    from . import background, boltzmann, bcm, \
+        cls, correlations, covariances, \
+        pk2d, power, tracers, halos, nl_pt
+    subs = [background, boltzmann, bcm, cls, correlations, covariances,
+            pk2d, power, tracers, halos, nl_pt]
+    funcs = [getmembers(sub, isfunction) for sub in subs]
+    funcs = [func for sub in funcs for func in sub]
+    for name, func in funcs:
+        pars = signature(func).parameters
+        if list(pars)[0] == "cosmo":
+            vars()[name] = func
+    del background, boltzmann, bcm, cls, correlations, covariances, \
+        pk2d, power, tracers, halos, nl_pt
+
```

Comment:
Can we check how much extra time all these new lines add to the creation of a standard cosmology?
I'd like to check this both on a laptop and on a server (@nikfilippas you can use glamdring).

Reply:
I've checked it with Spyder's built-in `%timeit` functionality, and, within the standard deviation, there is no difference at all.
That was with a loop of 70,000 instantiations.

Reply:
- with: 42.3 µs ± 135 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
- without: 40.1 µs ± 198 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

Reply:
ok, good

Reply:
@nikfilippas did you test this on glamdring?

Reply:
Timings on glamdring:
- with: `37.0 μs per loop (10000 loops)`
- without: `37.3 μs per loop (10000 loops)`

No statistically significant difference then.